### ISSUE TITLE
Panic when `user.name` or `user.email` contains `<`, `>`, or `\n`

### ISSUE THREAD

**Issue by martinvonz** [jj-vcs/jj#4865]
## Steps to Reproduce the Problem

```fish
cd (mktemp -d)
jj git init
jj config set --repo user.name "0 < 1"
jj new
```

## Expected Behavior

No crash.

## Actual Behavior

```
thread 'main' panicked at third_party/rust/jj_lib/v0_23/src/git_backend.rs:1265:44:
called `Result::unwrap()` on an `Err` value: Custom { kind: Other, error: IllegalCharacter }
```

The error comes from [here in Gitoxide](https://github.com/GitoxideLabs/gitoxide/blob/66c222c255b05ef8ff9b43609dcbf8b4ca00e01a/gix-actor/src/signature/mod.rs#L85-L86). The error should be propagated instead of resulting in a panic in Gitoxide [here](https://github.com/GitoxideLabs/gitoxide/blob/66c222c255b05ef8ff9b43609dcbf8b4ca00e01a/gix/src/repository/object.rs#L167) . I'll file a separate bug against Gitoxide for that.

## Specifications

- Platform: Linux (but presumably happens on all)
- Version: 0.23.0

**Comment by ilyagr** [comment]
"Hi, my name is <!#><$$%<!&. Resistance is futile"

I'd panic too.

**Comment by martinvonz** [comment]
https://github.com/GitoxideLabs/gitoxide/issues/1678 for the Gitoxide bug

**Comment by barryoneill** [comment]
It's possible this is unrelated, but I'm also having trouble setting my name:

```
❯ jj config set --user user.name "Barry O'Neill"
error: invalid value 'Barry O'Neill' for '<VALUE>': TOML parse error at line 1, column 1
  |
1 | Barry O'Neill
  | ^
invalid string
expected `"`, `'`
```

If I drop the apostrophe, it works fine ( But then that isn't my name 😄  )

**Comment by martinvonz** [comment]
Yes, it's unrelated. You need to quote it as a TOML string because of the apostrophe. That would be `"Barry O'Neill"`. Then you need to shell escape that too. If you use Bash, you can do `'"Barry O'\''Neill"'`. If you use Fish, you can do `'"Barry O\'Neill"'`. We should clarify that with an example in the `jj config set --help` text.

**Comment by martinvonz** [comment]
Hopefully #5619 would have helped you, @barryoneill

**Comment by barryoneill** [comment]
Yes, that helped.  I don't think it's ideal to expose end users to the underlying encoding of the config (it's been a while since a tool required me to escape my name), but I appreciate the workaround. Thanks!

**Comment by steadmon** [comment]
[https://github.com/GitoxideLabs/gitoxide/issues/1678](https://github.com/GitoxideLabs/gitoxide/issues/1678) was fixed in gix v0.68.0, and we're on v0.70.0 now. I just tried the repro case above, and it was successful, so I'll close this issue.
